kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -133,15 +132,27 @@ public void flush(Map<TopicPartition, OffsetAndMetadata> offsets) {
     topicPartitionManager.resumeAll();
   }
 
-  private PartitionedTableId getRecordTable(SinkRecord record) {
-    // Dynamically update topicToBaseTableIds mapping. topicToBaseTableIds was used to be
-    // constructed when connector starts hence new topic configuration needed connector to restart.
-    // Dynamic update shall not require connector restart and shall compute table id in runtime.
-    if (!topicsToBaseTableIds.containsKey(record.topic())) {
-      TopicToTableResolver.updateTopicToTable(config, record.topic(), topicsToBaseTableIds);
+  private void ensureExistingTable(TableId table) {
```

Comment:
We could rename this to `maybeEnsureExistingTable` and then do the check for `config.getBoolean(config.TABLE_CREATE_CONFIG)` here instead of putting that burden on the caller.